src/main/java/de/dennisguse/opentracks/content/TrackDataHub.java

```diff
@@ -71,12 +71,12 @@
     private final ContentProviderUtils contentProviderUtils;
     private final int targetNumPoints;
 
-    private SharedPreferences sharedPreferences;
-
     private boolean started;
     private HandlerThread handlerThread;
     private Handler handler;
 
+    private TrackRecordingServiceConnection trackRecordingServiceConnection;
```

Comment:
Instantiate here instead of constructor + final

Reply:
I cannot to do that because I get a illegal forwared reference to bindService (or I don't know how to do it :thinking: )

Reply:
An attribute can only be referenced after it was declared; basically move this line after `private final Runnable bindCallback = () -> {`

Reply:
If I move it after bindCallback definition then I get an illegal forwarded reference to trackRecordingServiceConnection.